### PR TITLE
Implemented consume flow

### LINKED ISSUE #490: Consume flow

### PR DESCRIPTION
**Pull request by mariacarmina**
Fixes #490  .

Changes proposed in this PR:

- implement `download` endpoint;
- create `test_consume_flow.py`
- add helper function for provider fees retrival
- fixed `_add_defaults` function from `OceanAssetsV4.create`.

### CODE REVIEW

**Comment by mariacarmina** [comment]
I marked as a draft until https://github.com/oceanprotocol/provider/pull/317 is merged, because I encountered this type of error and `documentId` appears to be `null` even if the sent payload contains the right `DID`
image

image

**Comment by alexcos20** [comment]
make sure that you have the latest images of ocean_contracts and provider in barge

File: ocean_lib/data_provider/data_service_provider.py [hunk 1]
```diff
@@ -145,6 +145,62 @@ def encrypt(
 
         return response
 
+    @staticmethod
+    @enforce_types
+    def download(
```
Comment (calina-c) [inline ocean_lib/data_provider/data_service_provider.py:150]: you already have download_service in provider. Adapt that one, please.

File: ocean_lib/ocean/v4/ocean_assets.py [hunk 2]
```diff
@@ -78,7 +78,7 @@ def _add_defaults(
 
         if not has_access_service:
             access_service = self.build_access_service(
-                service_id="1",
+                service_id="0",
```
Comment (calina-c) [inline ocean_lib/ocean/v4/ocean_assets.py:81]: why?

Comment (mariacarmina) [reply ocean_lib/ocean/v4/ocean_assets.py:81]: I got an error from provider that the requested service needs to have the id set to 0 instead of 1.

**Comment by mariacarmina** [comment]
The file downloaded into `consume-downloads`
Screenshot from 2022-01-04 13-18-22

File: ocean_lib/ocean/v4/ocean_assets.py (hunk 2 above)
Comment (mariacarmina) [reply ocean_lib/ocean/v4/ocean_assets.py:81]: It was my bad when I first implemented this part.

**Comment by mariacarmina** [comment]
Due to the conflicts from merging #627  and other comments that needed to be fixed, I have created #636.

**Pull request closed without merging**